README.md

```diff
@@ -47,42 +47,74 @@ CMD ["./guestbook"]
 EXPOSE 3000
 ```
 
-Docker files specific a set of commands, at each command we build a
-new layer.
+A Dockerfile contains a set of image build directives, such as `COPY`
+or `RUN`. When each directive is executed it creates a new layer in
+the image repository. Future layers build on the ones before it. This
+layered approach creates some efficiencies for storing and fetching
+image content.
 
-FROM - pulls a layer from a remote repository. The default
-repository is the docker image registry. This could be a complete OS
-image, or some partial image that used to do something small. In this
-example we pull in 2 different base images. One is a named layer for
-the golang build environment, so that it can be referenced later. The
-second one is the base image with just enough operating system to
-provide debug facilities inside the image.
+## Directives
 
-RUN - executes a command, all output files go into the layer. There
+Here is a brief overview of the directives used above.
+
+`FROM` - pulls a layer from a remote repository. The default
+repository is the docker image registry. A special form of `FROM` uses
+the `as` keyword to create a named building space.
+
+`RUN` - executes a command, all output files go into the layer. There
 are a few `go get` commands that pull in libraries for the main
 application. Later the main.go is compiled using this.
 
-COPY - copy a file from the local filesystem into the container. This
+`COPY` - copy a file from the local filesystem into the container. This
 can work on a set of files.
 
-WORKDIR - set the working directory for the entrypoint
+`WORKDIR` - like unix `cwd`, change the workding directory. The
+impacts all future commands in the `Dockerfile`.
 
-CMD - what command should be run in this container if no exec is
-passed into it. CMD means that the command run is replaceable. There
+`CMD` - what command should be run in this container if no exec is
+passed into it. `CMD` means that the command run is replaceable. There
 is another stanza `ENTRYPOINT` which is not replaceable.
 
-EXPOSE - open up a specific port from this container to the
-infrastructure. Without this no network traffic will be allowed into
-the container.
+`EXPOSE` - open up a specific port from this container to the
```

Comment:
I think its more accurate to say: specifies which ports to automatically map to host ports. Requires the `-P` option on the `docker run` command.

Technically you can still open up ports w/o this using `-p`, this is covers the `-P` use case.